### PR TITLE
Issue 44 project b integration

### PR DESCRIPTION
**Pull request by shayansiddiqui**
integrated group 4 project. Need to merge after issue 43 is merged

### CODE REVIEW

**Review by iraklitchedia — DISMISSED** [review]
Looks good

File: UI/src/app/home/home.component.html [hunk 1]
```diff
@@ -1,24 +1,58 @@
...
+
+    </div>
+    <div class="col-4" style="margin-right: 5px">
+      <div class="row">
+        <app-text-input #textInput
+                        id="text-input"
+                        (plot)="plotGraph()">
+        </app-text-input>
+      </div>
+      <div class="row" style="margin-top: 20px">
```
Comment (csehlke) [inline UI/src/app/home/home.component.html]: that is part of the id buttons in the css file

File: UI/src/app/home/home.component.html [hunk 2]
```diff
@@ -1,24 +1,58 @@
...
+                [disabled]="shouldDisablePlotButton()">
+          <i class="fa fa-chart-line"></i> Plot
+        </button>
+      </div>
+    </div>
+    <div class="col-4" style="margin-left: 5px">
+      <div class="row">
+        <app-text-input #textInputUpdate
+                        id="text-input-update"
+                        (plot)="updateGraph()">
```
Comment (csehlke) [inline UI/src/app/home/home.component.html]: we should create a new `@Input()` to set the default text

File: UI/src/app/services/backend-connector.service.ts [hunk 3]
```diff
@@ -1,192 +1,203 @@
...
-    private parseLayout(data: Layout){
-        const map = new Map<number, Layout>();
-        map.set(0, data);
-        this._layout.next(map);
-    }
+        return this.http.post('https://js2018-group4.azurewebsites.net/api/nlp',
+            {
+                sentence: userInput,
+                data: data.traces,
+                layout: data.layout
```
Comment (csehlke) [inline UI/src/app/services/backend-connector.service.ts:54]: Did someone test this? I am pretty sure we need to manipulate the header here a bit.

File: UI/src/app/home/home.component.html [hunk 4]
```diff
@@ -1,24 +1,58 @@
...
+        </app-text-input>
+      </div>
+      <div class="row" style="margin-top: 20px">
+        <button id="updateButton"
+                class="btn btn-md btn-warning"
+                type="button"
+                (click)="updateGraph()"
+                [disabled]="shouldDisablePlotButton()">
```
Comment (csehlke) [inline UI/src/app/home/home.component.html]: we sould write a different methode to disable this button

File: UI/src/app/home/home.component.html [hunk 5]
```diff
@@ -1,24 +1,58 @@
...
-            type="button"
-            (click)="plotGraph()"
-            [disabled]="shouldDisablePlotButton()">
-      <i class="fa fa-chart-line"></i> Plot
-    </button>
+  <hr>
+  <div id="buttons" class="row" style="padding: 20px">
+
+
     <button id="resetButton"
```
Comment (csehlke) [inline UI/src/app/home/home.component.html]: the disable methode is not up to date anymore

**Review by csehlke — DISMISSED** [review]

File: UI/src/app/services/backend-connector.service.ts (hunk 3 above)
Comment (shayansiddiqui) [reply UI/src/app/services/backend-connector.service.ts:54]: Ya I tested it.. worked fine. I also sent you a screenshot

**Review by jyotirmay123 — APPROVED** [review]
looks good.

**Pull request merged**